pkg/cli/admin/nodeimage/basenodeimagecommand.go

```diff
@@ -87,9 +89,48 @@ func (c *BaseNodeImageCommand) getNodeJoinerPullSpec(ctx context.Context) error
 	}
 
 	// Extract the baremetal-installer image pullspec, since it
-	// provide the node-joiner tool.
+	// provides the node-joiner tool.
+
+	// First attempt to get the installer image from the configMap created by installer. This will work in disconnected environments.
+	installerImagesConfigMap, err := c.Client.CoreV1().ConfigMaps("openshift-config").Get(ctx, "installer-images", metav1.GetOptions{})
+	if err == nil {
+		images := installerImagesConfigMap.Data["images.json"]
+		if len(images) > 0 {
+			imageMap := make(map[string]string)
+			err := yaml.Unmarshal([]byte(images), &imageMap)
+			if err == nil {
+				installer, exists := imageMap["installer"]
+				if exists {
+					// Found pullSpec from configMap
+					c.log("installer pullspec obtained from installer-images configMap %s", installer)
+					c.nodeJoinerImage = installer
+					return nil
+				}
+			}
+		}
+	} else if !kapierrors.IsNotFound(err) {
+		c.log("failure retrieving installer-images configMap %s", err.Error())
+	} else {
```

Comment:
On reflection, maybe just put this log outside of an `else` block, since there are other ways to get to the fallback code (e.g. ConfigMap has been incorrectly manually created on an old version, so it exists but does not have the expected keys).

Reply:
Sure, I'll move it so we log just one failure retrieving installer-images configMap.